src/modules/75-cf/pages/feature-flags/__tests__/FeatureFlagsPage3.test.tsx

```diff
@@ -127,31 +108,38 @@ describe('FeatureFlagsPage', () => {
       useDeleteFeatureFlag: () => ({ mutate })
     })
 
-    mockImport('services/cd-ng', {
-      useGetEnvironmentListForProject: () => ({
-        data: mockEnvironments,
-        loading: false,
-        error: undefined,
-        refetch: jest.fn()
-      })
-    })
-
-    const { container } = render(
-      <TestWrapper
-        path="/account/:accountId/cf/orgs/:orgIdentifier/projects/:projectIdentifier/feature-flags"
-        pathParams={{ accountId: 'dummy', orgIdentifier: 'dummy', projectIdentifier: 'dummy' }}
-      >
-        <FeatureFlagsPage />
-      </TestWrapper>
-    )
+    renderComponent()
 
-    fireEvent.click(container.querySelector('[role="row"]:not(:first-of-type) [data-icon="Options"]') as HTMLElement)
+    fireEvent.click(document.querySelector('[role="row"]:not(:first-of-type) [data-icon="Options"]') as HTMLElement)
     fireEvent.click(document.querySelector('[icon="trash"]') as HTMLElement)
 
     fireEvent.click(document.querySelector('button[class*=intent-danger]') as HTMLButtonElement)
     await waitFor(() => expect(mutate).toBeCalledTimes(1))
   })
 
+  describe('FilterCards', () => {
+    test('should not render if there is no active Environment', async () => {
+      renderComponent()
+
+      expect(screen.queryAllByTestId('filter-card').length).toBe(0)
+    })
+    test('should render when Feature Flags exist but no active Environment', async () => {
+      renderActiveEnvironmentHook()
+      renderComponent()
+
+      expect(screen.queryAllByTestId('filter-card').length).toBe(6)
+    })
+    test('should not render if there is an active Environment but no flags', async () => {
+      mockImport('services/cf', {
+        useGetAllFeatures: () => ({ data: undefined, refetch: jest.fn() })
+      })
+      renderComponent()
+
+      expect(screen.queryAllByTestId('filter-card').length).toBe(0)
```

Comment:
There's a nicer assertion available for lengths
```suggestion
      expect(screen.queryAllByTestId('filter-card')).toHaveLength(0)
```